app/views/posts/_expanded.html.erb

```diff
@@ -293,6 +293,33 @@
           </div>
         <% end %>
 
+        <% if is_question && post.children.length >= SiteSetting["TableOfContentsThreshold"] %>
+          <div class="toc" id="toc-example-toggle">
+            <button class="toc--header" data-toggle="#toc-example-toggle" data-toggle-property="class" data-toggle-value="is-active">Table of Contents</button>
+            <% post.children.each do |answer| %>
+              <a href="#" class="toc--entry">
+                <div class="toc--badge">
+                  <span class="badge is-tag is-green">+<%= answer.upvote_count %></span>
+                </div>
+                <div class="toc--badge">
+                  <span class="badge is-tag is-red">&minus;<%= answer.downvote_count %></span>
+                </div>
+                <div class="toc--full">
+                  <% sanitised = raw(sanitize(answer.body, scrubber: scrubber)) %>
+                  <% max_length = SiteSetting["TableOfContentsExcerptLength"] %>
+                  <p>
+                    <%=raw sanitised[ 0 ... max_length ] %>
+                    <% if sanitised.length > max_length %>
+                      ...
+                    <% end %>
+                  </p>
+                  <p class="has-font-size-small"><%= time_ago_in_words(post.last_edited_at, locale: :en_abbrev) %> ago</p>
```

Comment:
[**issue:**](https://conventionalcomments.org/) This won't work on most setups, because the last_edited_at value is only set for some posts with recent edits, because it's only been added recently (for the "updated X days ago by" labels) and is hence limited to some specific actions.

Consider using `updated_at` here.

Reply:
@luap42 what would it take to eliminate that inconsistency?  Would a run through the DB to set a missing last_edited_at to the value of whatever that other field is fix this once and for all?

Reply:
@cellio I think it may be possible by setting last_edited_at to updated_at,. Although it is likely hard to tell, whether that's what we want, because last_edited_at implies something different, namely a proper post edit, while updated_at and last_activity have different meanings (updated_at: the database entry changed, last_activity: something happened to the post or its children). And I'm not sure whether anything else bases on last_edited_at being set correctly.

Reply:
@luap42 oh I see.  I misunderstood.  Yeah, if it's not completely clear and unambiguous, better to leave the DB alone and adjust the code.